lib/postgres/repository/order.go

```diff
@@ -0,0 +1,46 @@
+package repository
+
+import (
+	"github.com/GrabItYourself/giys-backend/lib/postgres/models"
+)
+
+func (r *Repository) GetOrderById(id string) (*models.Order, error) {
+	var order *models.Order
+	if err := r.pg.Where("id = ?", id).First(order).Error; err != nil {
+		return nil, err
+	}
+	return order, nil
+}
+
+func (r *Repository) CreateOrder(order *models.Order) error {
+	if err := r.pg.Create(&models.Order{
+		UserId: order.UserId,
+		ShopId: order.ShopId,
+		Status: order.Status,
+	}).Error; err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *Repository) UpdateOrder(order *models.Order) error {
+	if err := r.pg.Save(order).Error; err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *Repository) UpdateOrderStatus(orderId int, shopId int, status models.OrderStatus) (*models.Order, error) {
+	var order models.Order
+	if err := r.pg.Model(&order).Where("id = ? AND shop_id = ?", orderId, shopId).Update("status", models.CompletedStatus).Error; err != nil {
```

Comment:
why is status hard coded to `models.CompletedStatus` instead of using the `status` in the arguments